### ISSUE TITLE
Library build output different since 5.22, variable undefined

### ISSUE THREAD

**Issue by depl0y** [webpack/webpack#12742]
# Bug report

**What is the current behavior?**

I build using the following output configuration:
```
output: {
	filename: './js/react/[name]-src.js',
	library: ['RF', '[name]'],
	libraryTarget: 'var',
	path: outputPath,
},
```

This results in a variable `RF` which is `undefined`.

The declaration of a property of RF now occurs using the following code (generated):

```
var RF;
(self["webpackChunkRF_name_"] = self["webpackChunkRF_name_"] || []).push([["audit-trail"], {
```

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**

In versions of webpack before v5.22 (up until 5.21.2), this resulted in an object 'RF' being declared, where (as in the example above) the following would be available `RF['audit-trail']`.

The same file resulted in this before 5.22
```
var RF;(RF = typeof RF === "undefined" ? {} : RF)["audit-trail"] =
	(self["webpackChunkRF_name_"] = self["webpackChunkRF_name_"] || []).push([["audit-trail"], {
```

**Other relevant information:**
webpack version:  5.22 (also tried 5.23)
Node.js version: 15.7.0
Operating System: Windows 10
Additional tools:

**Comment by sokra** [comment]
Yes that seem to be a regression.

**Comment by evazion** [comment]
I believe I'm seeing the same bug. In my case `git bisect` indicates https://github.com/webpack/webpack/commit/9f3950e571c9e56ac6264d55ea31bac99a113fe9 is the broken commit.

**Comment by sokra** [comment]
I'm working on that, but it needs some refactoring...

**Comment by sokra** [comment]
fixed by #12836
